pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java

```diff
@@ -314,8 +316,16 @@ void refreshSegmentStates() {
       }
       if (!enabledCandidates.isEmpty()) {
         instanceCandidatesMap.put(segment, enabledCandidates);
+      } else {
+        // Do not count new segment as unavailable
+        List<String> candidateInstances = new ArrayList<>(candidates.size());
+        for (SegmentInstanceCandidate candidate : candidates) {
+          candidateInstances.add(candidate.getInstance());
+        }
+        LOGGER.info("Failed to find servers hosting new segment: {} for table: {} "
+                + "(all candidate instances: {} are disabled, not counting segment as unavailable)", segment,
```

Comment:
same here.
`all candidate instances: {} are disabled, but counting segment as available for new segment`

Reply:
Same as above, but revised the log to be easier to differentiate with old segment one